File: csharp/clock/Clock.cs
public class Clock : IEquatable<Clock>
{
    const int minutes_per_hour = 60;
    const int hours_per_cycle = 24;
    int hours;
    int minutes;
    public Clock(int hours, int minutes)
    {
        this.hours = hours;
        this.minutes = minutes;
        Sanitize();
    }

    public Clock Add(int minutesToAdd)
    {
        this.minutes += minutesToAdd;
        Sanitize();
        return this;
    }

    public bool Equals(Clock? other)
    {
        if (other == null)
            return false;

        return ((other.hours == this.hours) && (other.minutes == this.minutes));
    }


    public Clock Subtract(int minutesToSubtract)
    {
        this.minutes -= minutesToSubtract;
        Sanitize();
        return this;
    }

    private void Sanitize()
    {
        while (minutes < 0)
        {
            hours--;
            minutes += minutes_per_hour;
        }

        while (minutes >= minutes_per_hour)
        {
            hours++;
            minutes -= minutes_per_hour;
        }

        while (hours < 0)
        {
            hours += hours_per_cycle;
        }

        while (hours >= hours_per_cycle)
        {
            hours -= hours_per_cycle;
        }
    }

    public override string? ToString()
    {
        return $"{hours:D2}:{minutes:D2}";
    }
}

File: go/clock/clock.go
package clock

import "fmt"

// Define the Clock type here.
type Clock struct {
	hours   int
	minutes int
}

func New(h, m int) Clock {
	return Clock{hours: h, minutes: m}.adjust()
}

func (c Clock) adjust() Clock {
	for c.minutes < 0 {
		c.minutes += 60
		c.hours--
	}
	for c.minutes >= 60 {
		c.minutes -= 60
		c.hours++
	}

	for c.hours < 0 {
		c.hours += 24
	}

	for c.hours >= 24 {
		c.hours -= 24
	}
	return c
}

func (c Clock) Add(m int) Clock {
	c.minutes += m
	return c.adjust()
}

func (c Clock) Subtract(m int) Clock {
	c.minutes -= m
	return c.adjust()
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hours, c.minutes)
}
